src/lights.js

```diff
@@ -1,33 +1,34 @@
-/**
- * Handles the hot updates
- * generated by webpack during development.
- */
-if (module.hot) {
-    // module.hot.accept(['./tree.js'], function() {
-    //    clearInterval(lightsInterval);
-    //    turnOn();
-    // });
-    // module.hot.accept();
-    module.hot.dispose(_data => {
-      clearInterval(lightsInterval);
-    });
-}
+
 
 /**
  * Changes the look of
  * some 'needles' in the tree
  * every 1000ms
  */
-let lightsInterval;
 export function turnOn(fir) {
     const blinkRate = 1000;
     const rowsCount = fir.rowsCount;
     const needles = fir.getNeedles();
 
-    lightsInterval = setInterval(() =>
+    const lightsInterval = setInterval(() =>
         blink(rowsCount, needles),
         blinkRate
     );
+    
+    /**
+     * Handles the hot updates
+     * generated by webpack during development.
+     */
+    if (module.hot) {
+        // module.hot.accept(['./tree.js'], function() {
+        //    clearInterval(lightsInterval);
+        //    turnOn();
+        // });
+        // module.hot.accept();
+        module.hot.dispose(_data => {
+          clearInterval(lightsInterval);
+        });
+    }
```

Comment:
The `dispose` event handler can be set inside the `turnOn`.  It will also avoid variable pollution.